ext/fetch/26_fetch.js

```diff
@@ -210,44 +212,60 @@
       const reader = reqBody.getReader();
       WeakMapPrototypeSet(requestBodyReaders, req, reader);
       (async () => {
-        while (true) {
-          const { value, done } = await PromisePrototypeCatch(
-            reader.read(),
-            (err) => {
-              if (terminator.aborted) return { done: true, value: undefined };
-              throw err;
-            },
-          );
+        let done = false;
+        while (!done) {
+          let val;
+          try {
+            const res = await reader.read();
+            done = res.done;
+            val = res.value;
+          } catch (err) {
+            if (terminator.aborted) break;
```

Comment:
You don't want to send an error to the response stream if terminated?

Reply:
Yes I do - if `terminator.aborted` is set, `core.shutdown` won't be called further down in the code.

Reply:
Consider using the same error paths, multiple paths lead to mistakes and difficult-to-understand code.

Reply:
Yeah, I want to refactor the terminator code paths in a follow up.